08-single-page-web-apps/mongo_mongoose/readme.md

```diff
@@ -480,129 +435,463 @@ kitty.save(function (err) {
+```
 
-**[Research](http://mongoosejs.com/docs/models.html):** How do I find or count restaurants with mongoose?
+In `controller/authorsController.js`:
 
+```js
+addReminder: function(req, res){
+  AuthorModel.findById(req.params.id, function(err, docs){
+    docs.reminders.push(new ReminderModel({body: req.body.body}))
+    docs.save(function(err){
+      if(!err){
+        res.redirect("/authors/" + req.params.id)
+      }
+    })
+  })
+}
+```
+
+> In this callback, we're finding an author by the id in the url. Then when we find it, we're going to push a new `Reminder` document to that author and save the author with that new reminder.
+
+Next, let's create the functionality to delete reminders as we complete them. Lets update our `views/authors/show.hbs` to include a form for deleting for each reminder:
+
+```
+{{#each reminders}}
+  <div class="reminders-show-each">
+    <p>{{body}}</p>
+    <form method="post" action="/authors/{{../_id}}/reminders/{{_id}}?_method=delete">
+        <input type="submit" value="Done!">
+    </form>
+  </div>
+{{/each}}
+```
+
+> The actions a little weird. In order to get the id of our parent object we use `../_id`. This will also be true of any other attributes we'd want from the parent object. Another weird thing is `?_method=delete`. This was the best work around we could find using `method-override` dependency to allow for `PUT` and `DELETE` requests in html forms on express.
+
+Now we just need set a route and a controller action.
 
-- `Restaurant.find({ conditions })`
-- `Restaurant.count({ conditions })`
+In `app.js`:
 
-``` js
-// List restaurants in zipcode 20001
-  var searchZipcode = 20001
-  Restaurant.count({"address.zipcode": searchZipcode}, function (err, restaurantCount) {
-    if (err) return handleError(err);
-    console.log("We have", restaurantCount, "in", searchZipcode)
+```js
+app.delete("/authors/:authorId/reminders/:id", authorsController.removeReminder)
+```
+
+In `controllers/authorsController.js`:
+
+```js
+removeReminder: function(req, res){
+  AuthorModel.findByIdAndUpdate(req.params.authorId, {
+    $pull:{
+      reminders: {_id: req.params.id}
+    }
+  }, function(err, docs){
+    if(!err){
+      res.redirect("/authors/" + req.params.authorId)
+    }
   })
+}
 ```
 
-## Exercise: Find restaurants by zip code
+> This is an alternate sytax `.findByIdAndUpdate()`. We can actually us this method and pass in an options object. In this case, we're setting a key of `$pull`. This will find the reminder by the id specified in the url(`req.params.id`) and get rid of it from the author. Then upon success of `.findByIdAndUpdate()` it redirects tot hat author's show page.
+
+## Authors Edit/Update/destroy
+Using what you've learned in class, set up edit update and destroy for authors.
 
-[X] 1. Define the Schema
-[X] 2. Instantiate a model.
-[ ] 3. Find restaurants.
-[ ] 4. Just list names.
-[ ] 5. List names & zipcodes (w/o _id)
```

Comment:
Looks like you have removed all the lists that periodically reminded us of the overview and the steps we were taking.  With all this code, I feel we need to work harder on keeping our attention focused on the steps.  On the reasons why we are writing all this code.